kiali.yaml

```diff
@@ -66,15 +66,117 @@ subpackages:
+
+        # Verify essential UI files exist
+        test -f /opt/kiali/console/manifest.json
+        test -f /opt/kiali/console/kiali_icon_lightbkg_16px.png
+    - uses: test/kwok/cluster
+    - name: Verify Kiali Health Endpoint
+      runs: |
+        # Get kubeconfig from Kwok cluster and write to file
+        kwokctl get kubeconfig > kubeconfig.yaml
+        if [ ! -s kubeconfig.yaml ]; then
+          echo "Error: Failed to get kubeconfig from Kwok cluster"
+          exit 1
+        fi
+
+        if ! kubectl --kubeconfig kubeconfig.yaml get nodes; then
+          echo "Error: Invalid kubeconfig"
+          cat kubeconfig.yaml
+          exit 1
+        fi
+
+        # Extract server URL from kubeconfig
+        SERVER_URL=$(kubectl --kubeconfig kubeconfig.yaml config view -o jsonpath='{.clusters[0].cluster.server}')
+        if [ -z "$SERVER_URL" ]; then
+          echo "Error: Could not extract server URL from kubeconfig"
+          exit 1
+        fi
+
+        # Parse host and port from server URL
+        KUBERNETES_SERVICE_HOST=$(echo $SERVER_URL | sed -E 's|https?://([^:]+):.*|\1|')
+        KUBERNETES_SERVICE_PORT=$(echo $SERVER_URL | sed -E 's|https?://[^:]+:([0-9]+).*|\1|')
+
+        echo "Setting KUBERNETES_SERVICE_HOST=$KUBERNETES_SERVICE_HOST"
+        echo "Setting KUBERNETES_SERVICE_PORT=$KUBERNETES_SERVICE_PORT"
+
+        # Create service account token directory and files
+        mkdir -p /var/run/secrets/kubernetes.io/serviceaccount
+        echo "dummy-token" > /var/run/secrets/kubernetes.io/serviceaccount/token
+        echo "dummy-namespace" > /var/run/secrets/kubernetes.io/serviceaccount/namespace
+        echo "dummy-ca.crt" > /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+
+        # Create minimal config with Kwok cluster settings
+        PORT=20001
+        cat >kiali-config.yaml <<EOF
+        server:
+          port: $PORT
+          address: ""  # Listen on all interfaces
+          gzip_enabled: true
+          cors_allow_all: false
+        auth:
+          strategy: anonymous
+        external_services:
+          istio:
+            root_namespace: istio-system
+        deployment:
+          accessible_namespaces:
+            - '**'
+        kubernetes:
+          config_file: "$(pwd)/kubeconfig.yaml"
+          in_cluster: false
+          service_account_token_path: "/var/run/secrets/kubernetes.io/serviceaccount/token"
+          service_account_cert_path: "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+          use_service_account: true
+        EOF
+
+        echo "Using kubeconfig at: $(pwd)/kubeconfig.yaml"
+        echo "Kiali config:"
+        cat kiali-config.yaml
+
+        # Start Kiali in background with config and environment variables
+        KUBERNETES_SERVICE_HOST=$KUBERNETES_SERVICE_HOST \
+        KUBERNETES_SERVICE_PORT=$KUBERNETES_SERVICE_PORT \
+        /opt/kiali/kiali --config kiali-config.yaml --kubeconfig kubeconfig.yaml &>kiali.log &
```

Comment:
Let's use `uses: test/daemon-check-output` here, and omit other `kill` life-cycle checks below. In the `post:` we can call `curl`.